homeassistant/components/onewire/config_flow.py

```diff
@@ -0,0 +1,196 @@
+        if user_input is not None:
+            self.onewire_config.update(user_input)
+            if CONF_TYPE_OWSERVER == user_input[CONF_TYPE]:
+                return await self.async_step_owserver()
+            if CONF_TYPE_SYSBUS == user_input[CONF_TYPE]:
+                return await self.async_step_mount_dir()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_USER,
+            errors=errors,
+        )
+
+    async def async_step_owserver(self, user_input=None):
+        """Handle OWServer configuration."""
+        errors = {}
+        if user_input:
+            # Prevent duplicate entries
+            if is_duplicate_owserver_entry(self.hass, user_input):
+                return self.async_abort(reason="already_configured")
+
+            self.onewire_config.update(user_input)
+
+            try:
+                info = await validate_input_owserver(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(
+                    title=info["title"], data=self.onewire_config
+                )
+
+        return self.async_show_form(
+            step_id="owserver",
+            data_schema=DATA_SCHEMA_OWSERVER,
+            errors=errors,
+        )
+
+    async def async_step_mount_dir(self, user_input=None):
+        """Handle SysBus configuration."""
+        errors = {}
+        if user_input:
+            # Prevent duplicate entries
+            await self.async_set_unique_id(
+                f"{CONF_TYPE_SYSBUS}:{user_input[CONF_MOUNT_DIR]}"
+            )
+            self._abort_if_unique_id_configured()
+
+            self.onewire_config.update(user_input)
+
+            try:
+                info = await validate_input_mount_dir(self.hass, user_input)
+            except InvalidPath:
+                errors["base"] = "invalid_path"
+            else:
+                return self.async_create_entry(
+                    title=info["title"], data=self.onewire_config
+                )
+
+        return self.async_show_form(
+            step_id="mount_dir",
+            data_schema=DATA_SCHEMA_MOUNTDIR,
+            errors=errors,
+        )
+
+    async def async_step_import(self, platform_config):
+        """Handle import configuration from YAML."""
+        # OWServer
+        if platform_config[CONF_TYPE] == CONF_TYPE_OWSERVER:
+            if CONF_PORT not in platform_config:
+                platform_config[CONF_PORT] = DEFAULT_OWSERVER_PORT
+            return await self.async_step_owserver(platform_config)
+
+        # SysBus
+        if platform_config[CONF_TYPE] == CONF_TYPE_SYSBUS:
+            if CONF_MOUNT_DIR not in platform_config:
+                platform_config[CONF_MOUNT_DIR] = DEFAULT_SYSBUS_MOUNT_DIR
+            return await self.async_step_mount_dir(platform_config)
+
+        # OWFS
+        else:  # pragma: no cover
```

Comment:
Just remove this and outdent below.

Reply:
But then I miss the 100% coverage on config-flow - I've added InvalidOneWireType

Reply:
Can't we add the coverage disable anywhere in the code above where we want it to stop?

Reply:
Updated again in commit 7973954
`# pragma: no cover` only seems to work on a "branch" (if/else) but I couldn't find to ignore all subsequent lines.

Instead I've just moved OWFS above SysBus and I think that did the trick

Reply:
Good!